Meta/Lagom/Tools/CodeGenerators/GMLCompiler/main.cpp

```diff
@@ -0,0 +1,321 @@
+            if (child_value.is_array())
+                // FIXME: propagate precise cause
+                return Error::from_string_view("Nested arrays are not supported"sv);
+
+#define HANDLE_TYPE(is_type, type_name)                                                             \
+    if (child_value.is_type() && (!child_type.has_value() || child_type.value() == #type_name##sv)) \
+        child_type = #type_name##sv;                                                                \
+    else
+
+            HANDLE_TYPE(is_string, StringView)
+            HANDLE_TYPE(is_integer<i64>, i64)
+            HANDLE_TYPE(is_integer<u64>, u64)
+            HANDLE_TYPE(is_bool, bool)
+            HANDLE_TYPE(is_double, double)
+            // FIXME: propagate precise error cause
+            return Error::from_string_view("Inconsistent contained type in JSON array"sv);
+#undef HANDLE_TYPE
+        }
+        // FIXME: propagate precise error cause
+        if (!child_type.has_value())
+            return Error::from_string_view("Empty JSON array; cannot deduce type."sv);
+
+        StringBuilder initializer;
+        initializer.appendff("Array<{}, {}> {{ "sv, child_type.release_value(), array.size());
+        for (auto const& child_value : array.values())
+            initializer.appendff("{}, ", TRY(generate_initializer_for(child_value)));
+        initializer.append("} "sv);
+        return initializer.to_string();
+    }
+    // FIXME: propagate precise cause
+    return Error::from_string_view("Unsupported JSON value"sv);
+}
+
+// Loads an object and assigns it to the RefPtr<Widget> variable named object_name.
+// All loading happens in a separate block.
+ErrorOr<void> generate_loader_for_object(GUI::GML::Object& gml_object, SourceGenerator generator, String object_name, size_t indentation, UseObjectConstructor use_object_constructor)
+{
+    generator.set("object_name", object_name.to_deprecated_string());
+    generator.set("class_name", gml_object.name());
+
+    auto append = [&]<size_t N>(auto& generator, char const(&text)[N]) -> ErrorOr<void> {
+        generator.append(TRY(String::repeated(' ', indentation * 4)).bytes_as_string_view());
+        generator.appendln(text);
+        return {};
+    };
+
+    generator.append(TRY(String::repeated(' ', (indentation - 1) * 4)).bytes_as_string_view());
+    generator.appendln("{");
+    if (use_object_constructor == UseObjectConstructor::Yes)
+        TRY(append(generator, "@object_name@ = TRY(@class_name@::try_create());"));
+    else
+        TRY(append(generator, "@object_name@ = TRY(try_make_ref_counted<@class_name@>());"));
+
+    // Properties
+    TRY(gml_object.try_for_each_property([&](StringView key, NonnullRefPtr<GUI::GML::JsonValueNode> value) -> ErrorOr<void> {
+        auto value_code = TRY(generate_initializer_for(value));
+        auto const& ui_dimension_properties = get_ui_dimension_properties();
+        if (ui_dimension_properties.contains(key)) {
```

Comment:
This would be nicer as
```suggestion
        if (is_ui_dimension_property(key)) {
```

where you rewrite your `get_ui_dimension_properties` function as
```c++
static bool is_ui_dimension_property(StringView key) {
    HashMap... properties;
    if (...) { ... }
    return properties.contains(key);
}
```